src/data/nat/cast/basic.lean

```diff
@@ -206,6 +206,15 @@ by { apply ne_zero.of_map f, simp only [map_nat_cast, hn] }
 
 end ring_hom_class
 
+protected theorem function.injective.char_zero' {M N F} [add_monoid_with_one M]
+  [add_monoid_with_one N] [char_zero M] [add_monoid_hom_class F M N] {f : F}
+  (hf : function.injective f) (h₁ : f 1 = 1) : char_zero N :=
+⟨λ m n h, nat.cast_injective $ hf $ by rwa [map_nat_cast' f h₁, map_nat_cast' f h₁]⟩
+
+protected theorem function.injective.char_zero {R S F} [non_assoc_semiring R] [non_assoc_semiring S]
```

Comment:
```suggestion
/-- For the version in the other direction, see `ring_hom.char_zero` -/
protected theorem function.injective.char_zero {R S F} [non_assoc_semiring R] [non_assoc_semiring S]
```
though this suggests this might have a better home at `ring_hom.char_zero_of_injective` or similar.

Reply:
I'm going to drop this PR and use `ring_hom.char_zero_iff` instead.